app/models/offer.rb

```diff
@@ -207,6 +208,43 @@ def nondraft_states
     def donor_states
       valid_states - ["draft"]
     end
+
+    def priority_and_non_priority_offers_count_for(self_reviewer: false)
```

Comment:
@abulsayyad123 Could you share a sample json output for the requests, I have trouble understanding the key renaming and all

Reply:
Sure,
```
{
  active_offers_total_count: 166
  priority_receiving: 2
  priority_reviewed: 127
  priority_scheduled: 13
  priority_under_review: 6
  receiving: 2
  reviewed: 127
  reviewer_active_offers_total_count: 4
  reviewer_priority_under_review: 2
  reviewer_under_review: 4
  scheduled: 13
  under_review: 24
}
```

Reply:
please change this method's name, as it is not just sending `priority_and_non_priority_offers_count_for` as json, but other parameters too.

Reply:
Instead of creating the hashes and then renaming keys, you can maybe pass a key prefix to the methods

pseudo code
```ruby
def counter_function(filters, key_prefix: '')
  # ... returns a hash
end

def caller_function
    res = {}
    res.merge! counter_function({ priority: false })
    res.merge! counter_function({ priority: true }, key_prefix: 'priority_')
    res
end

```

Reply:
Also maybe the states could be passed as a be passed as a parameter. Currently the requirements are for `active_states`, but that might change in the future. 

To extend the pseudo code from above :

```ruby


def counter_function(filters, key_prefix: '')
  # ... returns a hash
end

def caller_function
    res = {}
    res.merge! counter_function({ states: ACTIVE_STATES, priority: false })
    res.merge! counter_function({ states: ACTIVE_STATES, priority: true }, key_prefix: 'priority_')
    res
end
```